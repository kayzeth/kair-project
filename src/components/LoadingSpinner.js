import React from 'react';

const LoadingSpinner = () => (
  <div className="loading-spinner" data-testid="loading-spinner">
    <div className="spinner"></div>
  </div>
);

export default LoadingSpinner;
