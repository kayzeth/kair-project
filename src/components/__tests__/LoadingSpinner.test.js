import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import LoadingSpinner from '../LoadingSpinner';

// Mock for react-router-dom as per project memory
jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
  useLocation: jest.fn().mockReturnValue({ pathname: '/' }),
  Link: ({ children, to }) => <a href={to}>{children}</a>
}));

describe('LoadingSpinner Component', () => {
  test('renders correctly', () => {
    render(<LoadingSpinner />);
    
    // Check that the loading spinner container is rendered
    const spinnerContainer = screen.getByTestId('loading-spinner');
    expect(spinnerContainer).toBeInTheDocument();
    expect(spinnerContainer).toHaveClass('loading-spinner');
    
    // Check that the spinner element is rendered
    const spinnerElement = spinnerContainer.querySelector('.spinner');
    expect(spinnerElement).toBeInTheDocument();
  });

  test('has the correct structure', () => {
    const { container } = render(<LoadingSpinner />);
    
    // Check the DOM structure
    expect(container.firstChild).toHaveClass('loading-spinner');
    expect(container.firstChild.firstChild).toHaveClass('spinner');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<LoadingSpinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
