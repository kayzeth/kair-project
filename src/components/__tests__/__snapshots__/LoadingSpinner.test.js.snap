// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading-spinner"
    data-testid="loading-spinner"
  >
    <div
      class="spinner"
    />
  </div>
</DocumentFragment>
`;
